### ISSUE TITLE
Provision public dashboards

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#83017]
### Discussed in https://github.com/grafana/grafana/discussions/71702

<div type='discussions-op-text'>

<sup>Originally posted by **EdwardKuenen** July 17, 2023</sup>
It seems to me that public dashboards can not be provisioned, but must be a manual and cumbersome process.

I would very like to provision public dashboards, just like normal dashboards. The path of the public dashboard should not be random, but a specified path to use the same path in more environments, like production and pre-production.</div>

**Comment by evictorero** [comment]
Closing since this was already [implemented](https://github.com/grafana/grafana/issues/71803)

**Comment by acontreras89** [comment]
Can someone point me to the documentation on how to provision public dashboards? I cannot find it anywhere.

**Comment by evictorero** [comment]
This is the [API](https://grafana.com/docs/grafana/latest/developers/http_api/dashboard_public/) documentation

**Comment by acontreras89** [comment]
Thanks @evictorero! I think I may have misunderstood something... By reading the title and description of the issue I though this could be achieved via [provisioning files](https://grafana.com/docs/grafana/latest/administration/provisioning/). Will this be added anytime soon?

**Comment by evictorero** [comment]
You are right, I'm confused now about the meaning of this feature request. We don't have plans to support public dashboards as provisioning files but I will open this issue again so we can track the community interest about it.

**Comment by acontreras89** [comment]
Well, I am most definitely interested :D in the meanwhile, do you think this could be achieved. through the API? I saw [a post in Grafana forums](https://community.grafana.com/t/provision-public-dashboards/88008/5) which looks promising, but can I use this API to make a dashboard public while keeping a known URL for it? Thanks in advance!

**Comment by evictorero** [comment]
> Well, I am most definitely interested :D in the meanwhile, do you think this could be achieved.through the API? I saw [a post in Grafana forums](https://community.grafana.com/t/provision-public-dashboards/88008/5) which looks promising, but can I use this API to make a dashboard public while keeping a known URL for it? Thanks in advance!

This message mentions the dashboards API to make a dashboard public without the ability to specify an `accessToken`. If you want to set a specific `uid` and `accessToken` you can use the public dashboards [API](https://grafana.com/docs/grafana/latest/developers/http_api/dashboard_public/)

**Comment by Daniel-Brosnan-Blazquez** [comment]
Any news on this?

**Comment by AlexProgrammerDE** [comment]
I am personally also interested in this feature. I currently provision my dashboards using file provisioning and want to provision my dashboards as a public link.
I'll probably have to use the api right now to set it as public after the dashboards were provisioned.

**Comment by mjp4** [comment]
I would also like to share my interest in this feature. My Grafana instance is currently deployed ephemerally in a single step with a helm chart and no persistent state. I am able to deploy my desired dashboard and data source config in this way, but want to expose a public dashboard. This is currently not possible without further interaction.

**Comment by balazspados** [comment]
I would also prefer to provision the public dashboard via config file instead of an API call. Also the public URL contains a generated unique ID, which means you need to know the explicit URL to access it.
But, if you deploy Grafana with a none persisted storage, every single time it generates a new UID, this new URL must be distributed to users...

**Comment by AlexProgrammerDE** [comment]
Please add this. I'll bump
